packages/ses/src/cauterize-property.js

```diff
@@ -60,6 +63,13 @@ export const cauterizeProperty = (
           return;
         }
       }
+      if (
+        typeof obj === 'function' &&
+        (prop === 'caller' || prop === 'arguments')
+      ) {
+        warn(`Tolerating undeletable ${subPath}`);
+        return;
+      }
```

Comment:
But you first need to cauterize them, like I do with `.prototype`. In this case, you'd also 
- set them to undefined
- make them non-writable non-configurable data properties
- check the sloppy mode cases where these properties would change their value to verify that Hermes obeys the non-writable, non-configurable data property stability restriction.

But it's likely worse than that. These properties are symptomatic of these function being defined as sloppy functions, which we simply need to prohibit all access to. Builtin functions are supposed to act like strict functions, including the absence of `.caller` and `.argument` properties. On what functions are you seeing these properties? What is different about how they are defined?

Reply:
> On what functions are you seeing these properties?

the functions in the `output after` list (in description), i've dug thru the list more thoroughly spotting it's `.caller`, `.arguments` _and_ `.prototype` still, here's the breakdown

<details>
  <summary>output after (detailed)</summary>

```console
  Promise .caller

  Removing intrinsics.Promise.caller
  Tolerating undeletable intrinsics.Promise.caller
  Removing intrinsics.Promise.arguments
  Tolerating undeletable intrinsics.Promise.arguments

  Removing intrinsics.Promise._l
  Removing intrinsics.Promise._m
  Removing intrinsics.Promise._n

  Promise .resolve

  Removing intrinsics.Promise.resolve.caller
  Tolerating undeletable intrinsics.Promise.resolve.caller
  Removing intrinsics.Promise.resolve.arguments
  Tolerating undeletable intrinsics.Promise.resolve.arguments
  Removing intrinsics.Promise.resolve.prototype
  Tolerating undeletable intrinsics.Promise.resolve.prototype === undefined
  
  Promise .all

  Removing intrinsics.Promise.all.caller
  Tolerating undeletable intrinsics.Promise.all.caller
  Removing intrinsics.Promise.all.arguments
  Tolerating undeletable intrinsics.Promise.all.arguments
  Removing intrinsics.Promise.all.prototype
  Tolerating undeletable intrinsics.Promise.all.prototype === undefined

  Promise .reject

  Removing intrinsics.Promise.reject.caller
  Tolerating undeletable intrinsics.Promise.reject.caller
  Removing intrinsics.Promise.reject.arguments
  Tolerating undeletable intrinsics.Promise.reject.arguments
  Removing intrinsics.Promise.reject.prototype
  Tolerating undeletable intrinsics.Promise.reject.prototype === undefined

  Promise .race

  Removing intrinsics.Promise.race.caller
  Tolerating undeletable intrinsics.Promise.race.caller
  Removing intrinsics.Promise.race.arguments
  Tolerating undeletable intrinsics.Promise.race.arguments
  Removing intrinsics.Promise.race.prototype
  Tolerating undeletable intrinsics.Promise.race.prototype === undefined

  lockdown

  Removing intrinsics.lockdown.caller
  Tolerating undeletable intrinsics.lockdown.caller
  Removing intrinsics.lockdown.arguments
  Tolerating undeletable intrinsics.lockdown.arguments
  Removing intrinsics.lockdown.prototype
  Tolerating undeletable intrinsics.lockdown.prototype === undefined

  harden

  Removing intrinsics.harden.caller
  Tolerating undeletable intrinsics.harden.caller
  Removing intrinsics.harden.arguments
  Tolerating undeletable intrinsics.harden.arguments
  Removing intrinsics.harden.prototype
  Tolerating undeletable intrinsics.harden.prototype === undefined

  % Inert Function
  
  Removing intrinsics.%InertFunction%.caller
  Tolerating undeletable intrinsics.%InertFunction%.caller
  Removing intrinsics.%InertFunction%.arguments
  Tolerating undeletable intrinsics.%InertFunction%.arguments

  % Inert Generator Function

  Removing intrinsics.%InertGeneratorFunction%.caller
  Tolerating undeletable intrinsics.%InertGeneratorFunction%.caller
  Removing intrinsics.%InertGeneratorFunction%.arguments
  Tolerating undeletable intrinsics.%InertGeneratorFunction%.arguments

  % Inert Async Function

  Removing intrinsics.%InertAsyncFunction%.caller
  Tolerating undeletable intrinsics.%InertAsyncFunction%.caller
  Removing intrinsics.%InertAsyncFunction%.arguments
  Tolerating undeletable intrinsics.%InertAsyncFunction%.arguments

  % Initial Date

  Removing intrinsics.%InitialDate%.caller
  Tolerating undeletable intrinsics.%InitialDate%.caller
  Removing intrinsics.%InitialDate%.arguments
  Tolerating undeletable intrinsics.%InitialDate%.arguments

  % Shared Date

  Removing intrinsics.%SharedDate%.caller
  Tolerating undeletable intrinsics.%SharedDate%.caller
  Removing intrinsics.%SharedDate%.arguments
  Tolerating undeletable intrinsics.%SharedDate%.arguments
  Removing intrinsics.%SharedDate%.now.caller
  Tolerating undeletable intrinsics.%SharedDate%.now.caller
  Removing intrinsics.%SharedDate%.now.arguments
  Tolerating undeletable intrinsics.%SharedDate%.now.arguments
  Removing intrinsics.%SharedDate%.now.prototype
  Tolerating undeletable intrinsics.%SharedDate%.now.prototype === undefined

  % Initial GetStackString

  Removing intrinsics.%InitialGetStackString%.caller
  Tolerating undeletable intrinsics.%InitialGetStackString%.caller
  Removing intrinsics.%InitialGetStackString%.arguments
  Tolerating undeletable intrinsics.%InitialGetStackString%.arguments
  Removing intrinsics.%InitialGetStackString%.prototype
  Tolerating undeletable intrinsics.%InitialGetStackString%.prototype === undefined

  % Initial InitialError

  Removing intrinsics.%InitialError%.caller
  Tolerating undeletable intrinsics.%InitialError%.caller
  Removing intrinsics.%InitialError%.arguments
  Tolerating undeletable intrinsics.%InitialError%.arguments
  Removing intrinsics.%InitialError%.stackTraceLimit<get>.caller
  Tolerating undeletable intrinsics.%InitialError%.stackTraceLimit<get>.caller
  Removing intrinsics.%InitialError%.stackTraceLimit<get>.arguments
  Tolerating undeletable intrinsics.%InitialError%.stackTraceLimit<get>.arguments
  Removing intrinsics.%InitialError%.stackTraceLimit<get>.prototype
  Tolerating undeletable intrinsics.%InitialError%.stackTraceLimit<get>.prototype === undefined
  Removing intrinsics.%InitialError%.stackTraceLimit<set>.caller
  Tolerating undeletable intrinsics.%InitialError%.stackTraceLimit<set>.caller
  Removing intrinsics.%InitialError%.stackTraceLimit<set>.arguments
  Tolerating undeletable intrinsics.%InitialError%.stackTraceLimit<set>.arguments
  Removing intrinsics.%InitialError%.stackTraceLimit<set>.prototype
  Tolerating undeletable intrinsics.%InitialError%.stackTraceLimit<set>.prototype === undefined
  Removing intrinsics.%InitialError%.captureStackTrace.caller
  Tolerating undeletable intrinsics.%InitialError%.captureStackTrace.caller
  Removing intrinsics.%InitialError%.captureStackTrace.arguments
  Tolerating undeletable intrinsics.%InitialError%.captureStackTrace.arguments
  Removing intrinsics.%InitialError%.captureStackTrace.prototype
  Tolerating undeletable intrinsics.%InitialError%.captureStackTrace.prototype === undefined
  Removing intrinsics.%InitialError%.prepareStackTrace<get>.caller
  Tolerating undeletable intrinsics.%InitialError%.prepareStackTrace<get>.caller
  Removing intrinsics.%InitialError%.prepareStackTrace<get>.arguments
  Tolerating undeletable intrinsics.%InitialError%.prepareStackTrace<get>.arguments
  Removing intrinsics.%InitialError%.prepareStackTrace<get>.prototype
  Tolerating undeletable intrinsics.%InitialError%.prepareStackTrace<get>.prototype === undefined
  Removing intrinsics.%InitialError%.prepareStackTrace<set>.caller
  Tolerating undeletable intrinsics.%InitialError%.prepareStackTrace<set>.caller
  Removing intrinsics.%InitialError%.prepareStackTrace<set>.arguments
  Tolerating undeletable intrinsics.%InitialError%.prepareStackTrace<set>.arguments
  Removing intrinsics.%InitialError%.prepareStackTrace<set>.prototype
  Tolerating undeletable intrinsics.%InitialError%.prepareStackTrace<set>.prototype === undefined

  % Shared Error

  Removing intrinsics.%SharedError%.caller
  Tolerating undeletable intrinsics.%SharedError%.caller
  Removing intrinsics.%SharedError%.arguments
  Tolerating undeletable intrinsics.%SharedError%.arguments
  Removing intrinsics.%SharedError%.stackTraceLimit<get>.caller
  Tolerating undeletable intrinsics.%SharedError%.stackTraceLimit<get>.caller
  Removing intrinsics.%SharedError%.stackTraceLimit<get>.arguments
  Tolerating undeletable intrinsics.%SharedError%.stackTraceLimit<get>.arguments
  Removing intrinsics.%SharedError%.stackTraceLimit<get>.prototype
  Tolerating undeletable intrinsics.%SharedError%.stackTraceLimit<get>.prototype === undefined
  Removing intrinsics.%SharedError%.stackTraceLimit<set>.caller
  Tolerating undeletable intrinsics.%SharedError%.stackTraceLimit<set>.caller
  Removing intrinsics.%SharedError%.stackTraceLimit<set>.arguments
  Tolerating undeletable intrinsics.%SharedError%.stackTraceLimit<set>.arguments
  Removing intrinsics.%SharedError%.stackTraceLimit<set>.prototype
  Tolerating undeletable intrinsics.%SharedError%.stackTraceLimit<set>.prototype === undefined
  Removing intrinsics.%SharedError%.prepareStackTrace<get>.caller
  Tolerating undeletable intrinsics.%SharedError%.prepareStackTrace<get>.caller
  Removing intrinsics.%SharedError%.prepareStackTrace<get>.arguments
  Tolerating undeletable intrinsics.%SharedError%.prepareStackTrace<get>.arguments
  Removing intrinsics.%SharedError%.prepareStackTrace<get>.prototype
  Tolerating undeletable intrinsics.%SharedError%.prepareStackTrace<get>.prototype === undefined
  Removing intrinsics.%SharedError%.prepareStackTrace<set>.caller
  Tolerating undeletable intrinsics.%SharedError%.prepareStackTrace<set>.caller
  Removing intrinsics.%SharedError%.prepareStackTrace<set>.arguments
  Tolerating undeletable intrinsics.%SharedError%.prepareStackTrace<set>.arguments
  Removing intrinsics.%SharedError%.prepareStackTrace<set>.prototype
  Tolerating undeletable intrinsics.%SharedError%.prepareStackTrace<set>.prototype === undefined
  Removing intrinsics.%SharedError%.captureStackTrace.caller
  Tolerating undeletable intrinsics.%SharedError%.captureStackTrace.caller
  Removing intrinsics.%SharedError%.captureStackTrace.arguments
  Tolerating undeletable intrinsics.%SharedError%.captureStackTrace.arguments
  Removing intrinsics.%SharedError%.captureStackTrace.prototype
  Tolerating undeletable intrinsics.%SharedError%.captureStackTrace.prototype === undefined

  % Shared Math

  Removing intrinsics.%SharedMath%.random.caller
  Tolerating undeletable intrinsics.%SharedMath%.random.caller
  Removing intrinsics.%SharedMath%.random.arguments
  Tolerating undeletable intrinsics.%SharedMath%.random.arguments
  Removing intrinsics.%SharedMath%.random.prototype
  Tolerating undeletable intrinsics.%SharedMath%.random.prototype === undefined

  % Initial RegExp

  Removing intrinsics.%InitialRegExp%.caller
  Tolerating undeletable intrinsics.%InitialRegExp%.caller
  Removing intrinsics.%InitialRegExp%.arguments
  Tolerating undeletable intrinsics.%InitialRegExp%.arguments

  % Initial Shared RegExp

  Removing intrinsics.%SharedRegExp%.caller
  Tolerating undeletable intrinsics.%SharedRegExp%.caller
  Removing intrinsics.%SharedRegExp%.arguments
  Tolerating undeletable intrinsics.%SharedRegExp%.arguments

  % Initial SharedSymbol

  Removing intrinsics.%SharedSymbol%.caller
  Tolerating undeletable intrinsics.%SharedSymbol%.caller
  Removing intrinsics.%SharedSymbol%.arguments
  Tolerating undeletable intrinsics.%SharedSymbol%.arguments

  % Inert Compartment

  Removing intrinsics.%InertCompartment%.caller
  Tolerating undeletable intrinsics.%InertCompartment%.caller
  Removing intrinsics.%InertCompartment%.arguments
  Tolerating undeletable intrinsics.%InertCompartment%.arguments

  % Number Prototype

  Removing intrinsics.%NumberPrototype%.toLocaleString.caller
  Tolerating undeletable intrinsics.%NumberPrototype%.toLocaleString.caller
  Removing intrinsics.%NumberPrototype%.toLocaleString.arguments
  Tolerating undeletable intrinsics.%NumberPrototype%.toLocaleString.arguments
  Removing intrinsics.%NumberPrototype%.toLocaleString.prototype
  Tolerating undeletable intrinsics.%NumberPrototype%.toLocaleString.prototype === undefined

  % Promise Prototype

  Removing intrinsics.%PromisePrototype%.then.caller
  Tolerating undeletable intrinsics.%PromisePrototype%.then.caller
  Removing intrinsics.%PromisePrototype%.then.arguments
  Tolerating undeletable intrinsics.%PromisePrototype%.then.arguments
  Removing intrinsics.%PromisePrototype%.then.prototype
  Tolerating undeletable intrinsics.%PromisePrototype%.then.prototype === undefined
  Removing intrinsics.%PromisePrototype%.catch.caller
  Tolerating undeletable intrinsics.%PromisePrototype%.catch.caller
  Removing intrinsics.%PromisePrototype%.catch.arguments
  Tolerating undeletable intrinsics.%PromisePrototype%.catch.arguments
  Removing intrinsics.%PromisePrototype%.catch.prototype
  Tolerating undeletable intrinsics.%PromisePrototype%.catch.prototype === undefined
  Removing intrinsics.%PromisePrototype%.finally.caller
  Tolerating undeletable intrinsics.%PromisePrototype%.finally.caller
  Removing intrinsics.%PromisePrototype%.finally.arguments
  Tolerating undeletable intrinsics.%PromisePrototype%.finally.arguments
  Removing intrinsics.%PromisePrototype%.finally.prototype
  Tolerating undeletable intrinsics.%PromisePrototype%.finally.prototype === undefined

  % String Prototype

  Removing intrinsics.%StringPrototype%.localeCompare.caller
  Tolerating undeletable intrinsics.%StringPrototype%.localeCompare.caller
  Removing intrinsics.%StringPrototype%.localeCompare.arguments
  Tolerating undeletable intrinsics.%StringPrototype%.localeCompare.arguments
  Removing intrinsics.%StringPrototype%.localeCompare.prototype
  Tolerating undeletable intrinsics.%StringPrototype%.localeCompare.prototype === undefined

  % Function Prototype

  Removing intrinsics.%FunctionPrototype%.toString.caller
  Tolerating undeletable intrinsics.%FunctionPrototype%.toString.caller
  Removing intrinsics.%FunctionPrototype%.toString.arguments
  Tolerating undeletable intrinsics.%FunctionPrototype%.toString.arguments
  Removing intrinsics.%FunctionPrototype%.toString.prototype
  Tolerating undeletable intrinsics.%FunctionPrototype%.toString.prototype === undefined

  % Compartment Prototype

  Removing intrinsics.%CompartmentPrototype%.globalThis<get>.caller
  Tolerating undeletable intrinsics.%CompartmentPrototype%.globalThis<get>.caller
  Removing intrinsics.%CompartmentPrototype%.globalThis<get>.arguments
  Tolerating undeletable intrinsics.%CompartmentPrototype%.globalThis<get>.arguments
  Removing intrinsics.%CompartmentPrototype%.globalThis<get>.prototype
  Tolerating undeletable intrinsics.%CompartmentPrototype%.globalThis<get>.prototype === undefined
  Removing intrinsics.%CompartmentPrototype%.name<get>.caller
  Tolerating undeletable intrinsics.%CompartmentPrototype%.name<get>.caller
  Removing intrinsics.%CompartmentPrototype%.name<get>.arguments
  Tolerating undeletable intrinsics.%CompartmentPrototype%.name<get>.arguments
  Removing intrinsics.%CompartmentPrototype%.name<get>.prototype
  Tolerating undeletable intrinsics.%CompartmentPrototype%.name<get>.prototype === undefined
  Removing intrinsics.%CompartmentPrototype%.evaluate.caller
  Tolerating undeletable intrinsics.%CompartmentPrototype%.evaluate.caller
  Removing intrinsics.%CompartmentPrototype%.evaluate.arguments
  Tolerating undeletable intrinsics.%CompartmentPrototype%.evaluate.arguments
  Removing intrinsics.%CompartmentPrototype%.evaluate.prototype
  Tolerating undeletable intrinsics.%CompartmentPrototype%.evaluate.prototype === undefined
  Removing intrinsics.%CompartmentPrototype%.module.caller
  Tolerating undeletable intrinsics.%CompartmentPrototype%.module.caller
  Removing intrinsics.%CompartmentPrototype%.module.arguments
  Tolerating undeletable intrinsics.%CompartmentPrototype%.module.arguments
  Removing intrinsics.%CompartmentPrototype%.module.prototype
  Tolerating undeletable intrinsics.%CompartmentPrototype%.module.prototype === undefined
  Removing intrinsics.%CompartmentPrototype%.import.caller
  Tolerating undeletable intrinsics.%CompartmentPrototype%.import.caller
  Removing intrinsics.%CompartmentPrototype%.import.arguments
  Tolerating undeletable intrinsics.%CompartmentPrototype%.import.arguments
  Removing intrinsics.%CompartmentPrototype%.load.caller
  Tolerating undeletable intrinsics.%CompartmentPrototype%.load.caller
  Removing intrinsics.%CompartmentPrototype%.load.arguments
  Tolerating undeletable intrinsics.%CompartmentPrototype%.load.arguments
  Removing intrinsics.%CompartmentPrototype%.importNow.caller
  Tolerating undeletable intrinsics.%CompartmentPrototype%.importNow.caller
  Removing intrinsics.%CompartmentPrototype%.importNow.arguments
  Tolerating undeletable intrinsics.%CompartmentPrototype%.importNow.arguments
  Removing intrinsics.%CompartmentPrototype%.importNow.prototype
  Tolerating undeletable intrinsics.%CompartmentPrototype%.importNow.prototype === undefined

```

</details>

in summary

- Promise .caller

- Promise .resolve (including .prototype)
- Promise .all (including .prototype)
- Promise .reject (including .prototype)
- Promise .race (including .prototype)
- lockdown (including .prototype)
- harden (including .prototype)

- % Inert Function
- % Inert Generator Function
- % Inert Async Function
- % Initial Date

- % Shared Date (including .prototype)

- % Initial GetStackString
- % Initial InitialError
- % Shared Error
- % Shared Math (including .prototype)
- % Initial RegExp
- % Initial Shared RegExp
- % Initial SharedSymbol
- % Inert Compartment
- % Number Prototype (including .prototype)
- % Promise Prototype (including .prototype)
- % String Prototype (including .prototype)
- % Function Prototype (including .prototype)
- % Compartment Prototype (including .prototype)

Reply:
> On what functions are you seeing these properties?

the functions in the `output after` list (in description), i've dug thru the list more thoroughly spotting it's `.caller`, `.arguments` _and_ `.prototype` still, here's the breakdown

<details>
  <summary>output after (detailed)</summary>

```console
  Promise .caller

  Removing intrinsics.Promise.caller
  Tolerating undeletable intrinsics.Promise.caller
  Removing intrinsics.Promise.arguments
  Tolerating undeletable intrinsics.Promise.arguments

  Removing intrinsics.Promise._l
  Removing intrinsics.Promise._m
  Removing intrinsics.Promise._n

  Promise .resolve

  Removing intrinsics.Promise.resolve.caller
  Tolerating undeletable intrinsics.Promise.resolve.caller
  Removing intrinsics.Promise.resolve.arguments
  Tolerating undeletable intrinsics.Promise.resolve.arguments
  Removing intrinsics.Promise.resolve.prototype
  Tolerating undeletable intrinsics.Promise.resolve.prototype === undefined
  
  Promise .all

  Removing intrinsics.Promise.all.caller
  Tolerating undeletable intrinsics.Promise.all.caller
  Removing intrinsics.Promise.all.arguments
  Tolerating undeletable intrinsics.Promise.all.arguments
  Removing intrinsics.Promise.all.prototype
  Tolerating undeletable intrinsics.Promise.all.prototype === undefined

  Promise .reject

  Removing intrinsics.Promise.reject.caller
  Tolerating undeletable intrinsics.Promise.reject.caller
  Removing intrinsics.Promise.reject.arguments
  Tolerating undeletable intrinsics.Promise.reject.arguments
  Removing intrinsics.Promise.reject.prototype
  Tolerating undeletable intrinsics.Promise.reject.prototype === undefined

  Promise .race

  Removing intrinsics.Promise.race.caller
  Tolerating undeletable intrinsics.Promise.race.caller
  Removing intrinsics.Promise.race.arguments
  Tolerating undeletable intrinsics.Promise.race.arguments
  Removing intrinsics.Promise.race.prototype
  Tolerating undeletable intrinsics.Promise.race.prototype === undefined

  lockdown

  Removing intrinsics.lockdown.caller
  Tolerating undeletable intrinsics.lockdown.caller
  Removing intrinsics.lockdown.arguments
  Tolerating undeletable intrinsics.lockdown.arguments
  Removing intrinsics.lockdown.prototype
  Tolerating undeletable intrinsics.lockdown.prototype === undefined

  harden

  Removing intrinsics.harden.caller
  Tolerating undeletable intrinsics.harden.caller
  Removing intrinsics.harden.arguments
  Tolerating undeletable intrinsics.harden.arguments
  Removing intrinsics.harden.prototype
  Tolerating undeletable intrinsics.harden.prototype === undefined

  % Inert Function
  
  Removing intrinsics.%InertFunction%.caller
  Tolerating undeletable intrinsics.%InertFunction%.caller
  Removing intrinsics.%InertFunction%.arguments
  Tolerating undeletable intrinsics.%InertFunction%.arguments

  % Inert Generator Function

  Removing intrinsics.%InertGeneratorFunction%.caller
  Tolerating undeletable intrinsics.%InertGeneratorFunction%.caller
  Removing intrinsics.%InertGeneratorFunction%.arguments
  Tolerating undeletable intrinsics.%InertGeneratorFunction%.arguments

  % Inert Async Function

  Removing intrinsics.%InertAsyncFunction%.caller
  Tolerating undeletable intrinsics.%InertAsyncFunction%.caller
  Removing intrinsics.%InertAsyncFunction%.arguments
  Tolerating undeletable intrinsics.%InertAsyncFunction%.arguments

  % Initial Date

  Removing intrinsics.%InitialDate%.caller
  Tolerating undeletable intrinsics.%InitialDate%.caller
  Removing intrinsics.%InitialDate%.arguments
  Tolerating undeletable intrinsics.%InitialDate%.arguments

  % Shared Date

  Removing intrinsics.%SharedDate%.caller
  Tolerating undeletable intrinsics.%SharedDate%.caller
  Removing intrinsics.%SharedDate%.arguments
  Tolerating undeletable intrinsics.%SharedDate%.arguments
  Removing intrinsics.%SharedDate%.now.caller
  Tolerating undeletable intrinsics.%SharedDate%.now.caller
  Removing intrinsics.%SharedDate%.now.arguments
  Tolerating undeletable intrinsics.%SharedDate%.now.arguments
  Removing intrinsics.%SharedDate%.now.prototype
  Tolerating undeletable intrinsics.%SharedDate%.now.prototype === undefined

  % Initial GetStackString

  Removing intrinsics.%InitialGetStackString%.caller
  Tolerating undeletable intrinsics.%InitialGetStackString%.caller
  Removing intrinsics.%InitialGetStackString%.arguments
  Tolerating undeletable intrinsics.%InitialGetStackString%.arguments
  Removing intrinsics.%InitialGetStackString%.prototype
  Tolerating undeletable intrinsics.%InitialGetStackString%.prototype === undefined

  % Initial InitialError

  Removing intrinsics.%InitialError%.caller
  Tolerating undeletable intrinsics.%InitialError%.caller
  Removing intrinsics.%InitialError%.arguments
  Tolerating undeletable intrinsics.%InitialError%.arguments
  Removing intrinsics.%InitialError%.stackTraceLimit<get>.caller
  Tolerating undeletable intrinsics.%InitialError%.stackTraceLimit<get>.caller
  Removing intrinsics.%InitialError%.stackTraceLimit<get>.arguments
  Tolerating undeletable intrinsics.%InitialError%.stackTraceLimit<get>.arguments
  Removing intrinsics.%InitialError%.stackTraceLimit<get>.prototype
  Tolerating undeletable intrinsics.%InitialError%.stackTraceLimit<get>.prototype === undefined
  Removing intrinsics.%InitialError%.stackTraceLimit<set>.caller
  Tolerating undeletable intrinsics.%InitialError%.stackTraceLimit<set>.caller
  Removing intrinsics.%InitialError%.stackTraceLimit<set>.arguments
  Tolerating undeletable intrinsics.%InitialError%.stackTraceLimit<set>.arguments
  Removing intrinsics.%InitialError%.stackTraceLimit<set>.prototype
  Tolerating undeletable intrinsics.%InitialError%.stackTraceLimit<set>.prototype === undefined
  Removing intrinsics.%InitialError%.captureStackTrace.caller
  Tolerating undeletable intrinsics.%InitialError%.captureStackTrace.caller
  Removing intrinsics.%InitialError%.captureStackTrace.arguments
  Tolerating undeletable intrinsics.%InitialError%.captureStackTrace.arguments
  Removing intrinsics.%InitialError%.captureStackTrace.prototype
  Tolerating undeletable intrinsics.%InitialError%.captureStackTrace.prototype === undefined
  Removing intrinsics.%InitialError%.prepareStackTrace<get>.caller
  Tolerating undeletable intrinsics.%InitialError%.prepareStackTrace<get>.caller
  Removing intrinsics.%InitialError%.prepareStackTrace<get>.arguments
  Tolerating undeletable intrinsics.%InitialError%.prepareStackTrace<get>.arguments
  Removing intrinsics.%InitialError%.prepareStackTrace<get>.prototype
  Tolerating undeletable intrinsics.%InitialError%.prepareStackTrace<get>.prototype === undefined
  Removing intrinsics.%InitialError%.prepareStackTrace<set>.caller
  Tolerating undeletable intrinsics.%InitialError%.prepareStackTrace<set>.caller
  Removing intrinsics.%InitialError%.prepareStackTrace<set>.arguments
  Tolerating undeletable intrinsics.%InitialError%.prepareStackTrace<set>.arguments
  Removing intrinsics.%InitialError%.prepareStackTrace<set>.prototype
  Tolerating undeletable intrinsics.%InitialError%.prepareStackTrace<set>.prototype === undefined

  % Shared Error

  Removing intrinsics.%SharedError%.caller
  Tolerating undeletable intrinsics.%SharedError%.caller
  Removing intrinsics.%SharedError%.arguments
  Tolerating undeletable intrinsics.%SharedError%.arguments
  Removing intrinsics.%SharedError%.stackTraceLimit<get>.caller
  Tolerating undeletable intrinsics.%SharedError%.stackTraceLimit<get>.caller
  Removing intrinsics.%SharedError%.stackTraceLimit<get>.arguments
  Tolerating undeletable intrinsics.%SharedError%.stackTraceLimit<get>.arguments
  Removing intrinsics.%SharedError%.stackTraceLimit<get>.prototype
  Tolerating undeletable intrinsics.%SharedError%.stackTraceLimit<get>.prototype === undefined
  Removing intrinsics.%SharedError%.stackTraceLimit<set>.caller
  Tolerating undeletable intrinsics.%SharedError%.stackTraceLimit<set>.caller
  Removing intrinsics.%SharedError%.stackTraceLimit<set>.arguments
  Tolerating undeletable intrinsics.%SharedError%.stackTraceLimit<set>.arguments
  Removing intrinsics.%SharedError%.stackTraceLimit<set>.prototype
  Tolerating undeletable intrinsics.%SharedError%.stackTraceLimit<set>.prototype === undefined
  Removing intrinsics.%SharedError%.prepareStackTrace<get>.caller
  Tolerating undeletable intrinsics.%SharedError%.prepareStackTrace<get>.caller
  Removing intrinsics.%SharedError%.prepareStackTrace<get>.arguments
  Tolerating undeletable intrinsics.%SharedError%.prepareStackTrace<get>.arguments
  Removing intrinsics.%SharedError%.prepareStackTrace<get>.prototype
  Tolerating undeletable intrinsics.%SharedError%.prepareStackTrace<get>.prototype === undefined
  Removing intrinsics.%SharedError%.prepareStackTrace<set>.caller
  Tolerating undeletable intrinsics.%SharedError%.prepareStackTrace<set>.caller
  Removing intrinsics.%SharedError%.prepareStackTrace<set>.arguments
  Tolerating undeletable intrinsics.%SharedError%.prepareStackTrace<set>.arguments
  Removing intrinsics.%SharedError%.prepareStackTrace<set>.prototype
  Tolerating undeletable intrinsics.%SharedError%.prepareStackTrace<set>.prototype === undefined
  Removing intrinsics.%SharedError%.captureStackTrace.caller
  Tolerating undeletable intrinsics.%SharedError%.captureStackTrace.caller
  Removing intrinsics.%SharedError%.captureStackTrace.arguments
  Tolerating undeletable intrinsics.%SharedError%.captureStackTrace.arguments
  Removing intrinsics.%SharedError%.captureStackTrace.prototype
  Tolerating undeletable intrinsics.%SharedError%.captureStackTrace.prototype === undefined

  % Shared Math

  Removing intrinsics.%SharedMath%.random.caller
  Tolerating undeletable intrinsics.%SharedMath%.random.caller
  Removing intrinsics.%SharedMath%.random.arguments
  Tolerating undeletable intrinsics.%SharedMath%.random.arguments
  Removing intrinsics.%SharedMath%.random.prototype
  Tolerating undeletable intrinsics.%SharedMath%.random.prototype === undefined

  % Initial RegExp

  Removing intrinsics.%InitialRegExp%.caller
  Tolerating undeletable intrinsics.%InitialRegExp%.caller
  Removing intrinsics.%InitialRegExp%.arguments
  Tolerating undeletable intrinsics.%InitialRegExp%.arguments

  % Initial Shared RegExp

  Removing intrinsics.%SharedRegExp%.caller
  Tolerating undeletable intrinsics.%SharedRegExp%.caller
  Removing intrinsics.%SharedRegExp%.arguments
  Tolerating undeletable intrinsics.%SharedRegExp%.arguments

  % Initial SharedSymbol

  Removing intrinsics.%SharedSymbol%.caller
  Tolerating undeletable intrinsics.%SharedSymbol%.caller
  Removing intrinsics.%SharedSymbol%.arguments
  Tolerating undeletable intrinsics.%SharedSymbol%.arguments

  % Inert Compartment

  Removing intrinsics.%InertCompartment%.caller
  Tolerating undeletable intrinsics.%InertCompartment%.caller
  Removing intrinsics.%InertCompartment%.arguments
  Tolerating undeletable intrinsics.%InertCompartment%.arguments

  % Number Prototype

  Removing intrinsics.%NumberPrototype%.toLocaleString.caller
  Tolerating undeletable intrinsics.%NumberPrototype%.toLocaleString.caller
  Removing intrinsics.%NumberPrototype%.toLocaleString.arguments
  Tolerating undeletable intrinsics.%NumberPrototype%.toLocaleString.arguments
  Removing intrinsics.%NumberPrototype%.toLocaleString.prototype
  Tolerating undeletable intrinsics.%NumberPrototype%.toLocaleString.prototype === undefined

  % Promise Prototype

  Removing intrinsics.%PromisePrototype%.then.caller
  Tolerating undeletable intrinsics.%PromisePrototype%.then.caller
  Removing intrinsics.%PromisePrototype%.then.arguments
  Tolerating undeletable intrinsics.%PromisePrototype%.then.arguments
  Removing intrinsics.%PromisePrototype%.then.prototype
  Tolerating undeletable intrinsics.%PromisePrototype%.then.prototype === undefined
  Removing intrinsics.%PromisePrototype%.catch.caller
  Tolerating undeletable intrinsics.%PromisePrototype%.catch.caller
  Removing intrinsics.%PromisePrototype%.catch.arguments
  Tolerating undeletable intrinsics.%PromisePrototype%.catch.arguments
  Removing intrinsics.%PromisePrototype%.catch.prototype
  Tolerating undeletable intrinsics.%PromisePrototype%.catch.prototype === undefined
  Removing intrinsics.%PromisePrototype%.finally.caller
  Tolerating undeletable intrinsics.%PromisePrototype%.finally.caller
  Removing intrinsics.%PromisePrototype%.finally.arguments
  Tolerating undeletable intrinsics.%PromisePrototype%.finally.arguments
  Removing intrinsics.%PromisePrototype%.finally.prototype
  Tolerating undeletable intrinsics.%PromisePrototype%.finally.prototype === undefined

  % String Prototype

  Removing intrinsics.%StringPrototype%.localeCompare.caller
  Tolerating undeletable intrinsics.%StringPrototype%.localeCompare.caller
  Removing intrinsics.%StringPrototype%.localeCompare.arguments
  Tolerating undeletable intrinsics.%StringPrototype%.localeCompare.arguments
  Removing intrinsics.%StringPrototype%.localeCompare.prototype
  Tolerating undeletable intrinsics.%StringPrototype%.localeCompare.prototype === undefined

  % Function Prototype

  Removing intrinsics.%FunctionPrototype%.toString.caller
  Tolerating undeletable intrinsics.%FunctionPrototype%.toString.caller
  Removing intrinsics.%FunctionPrototype%.toString.arguments
  Tolerating undeletable intrinsics.%FunctionPrototype%.toString.arguments
  Removing intrinsics.%FunctionPrototype%.toString.prototype
  Tolerating undeletable intrinsics.%FunctionPrototype%.toString.prototype === undefined

  % Compartment Prototype

  Removing intrinsics.%CompartmentPrototype%.globalThis<get>.caller
  Tolerating undeletable intrinsics.%CompartmentPrototype%.globalThis<get>.caller
  Removing intrinsics.%CompartmentPrototype%.globalThis<get>.arguments
  Tolerating undeletable intrinsics.%CompartmentPrototype%.globalThis<get>.arguments
  Removing intrinsics.%CompartmentPrototype%.globalThis<get>.prototype
  Tolerating undeletable intrinsics.%CompartmentPrototype%.globalThis<get>.prototype === undefined
  Removing intrinsics.%CompartmentPrototype%.name<get>.caller
  Tolerating undeletable intrinsics.%CompartmentPrototype%.name<get>.caller
  Removing intrinsics.%CompartmentPrototype%.name<get>.arguments
  Tolerating undeletable intrinsics.%CompartmentPrototype%.name<get>.arguments
  Removing intrinsics.%CompartmentPrototype%.name<get>.prototype
  Tolerating undeletable intrinsics.%CompartmentPrototype%.name<get>.prototype === undefined
  Removing intrinsics.%CompartmentPrototype%.evaluate.caller
  Tolerating undeletable intrinsics.%CompartmentPrototype%.evaluate.caller
  Removing intrinsics.%CompartmentPrototype%.evaluate.arguments
  Tolerating undeletable intrinsics.%CompartmentPrototype%.evaluate.arguments
  Removing intrinsics.%CompartmentPrototype%.evaluate.prototype
  Tolerating undeletable intrinsics.%CompartmentPrototype%.evaluate.prototype === undefined
  Removing intrinsics.%CompartmentPrototype%.module.caller
  Tolerating undeletable intrinsics.%CompartmentPrototype%.module.caller
  Removing intrinsics.%CompartmentPrototype%.module.arguments
  Tolerating undeletable intrinsics.%CompartmentPrototype%.module.arguments
  Removing intrinsics.%CompartmentPrototype%.module.prototype
  Tolerating undeletable intrinsics.%CompartmentPrototype%.module.prototype === undefined
  Removing intrinsics.%CompartmentPrototype%.import.caller
  Tolerating undeletable intrinsics.%CompartmentPrototype%.import.caller
  Removing intrinsics.%CompartmentPrototype%.import.arguments
  Tolerating undeletable intrinsics.%CompartmentPrototype%.import.arguments
  Removing intrinsics.%CompartmentPrototype%.load.caller
  Tolerating undeletable intrinsics.%CompartmentPrototype%.load.caller
  Removing intrinsics.%CompartmentPrototype%.load.arguments
  Tolerating undeletable intrinsics.%CompartmentPrototype%.load.arguments
  Removing intrinsics.%CompartmentPrototype%.importNow.caller
  Tolerating undeletable intrinsics.%CompartmentPrototype%.importNow.caller
  Removing intrinsics.%CompartmentPrototype%.importNow.arguments
  Tolerating undeletable intrinsics.%CompartmentPrototype%.importNow.arguments
  Removing intrinsics.%CompartmentPrototype%.importNow.prototype
  Tolerating undeletable intrinsics.%CompartmentPrototype%.importNow.prototype === undefined

```

</details>

in summary

- Promise .caller
- Promise .resolve (including .prototype)
- Promise .all (including .prototype)
- Promise .reject (including .prototype)
- Promise .race (including .prototype)
- lockdown (including .prototype)
- harden (including .prototype)
- % Inert Function
- % Inert Generator Function
- % Inert Async Function
- % Initial Date
- % Shared Date (including .prototype)
- % Initial GetStackString
- % Initial InitialError (including .prototype with stack traces)
- % Shared Error (including .prototype with stack traces)
- % Shared Math (including .prototype)
- % Initial RegExp
- % Initial Shared RegExp
- % Initial SharedSymbol
- % Inert Compartment
- % Number Prototype (including .prototype)
- % Promise Prototype (including .prototype)
- % String Prototype (including .prototype)
- % Function Prototype (including .prototype)
- % Compartment Prototype (including .prototype)

Reply:
> What is different about how they are defined?

should hopefully figure this soon from Hermes source code

Reply:
At least in the case of the `Promise` subtree, I have an answer: https://github.com/facebook/hermes/issues/1009#issuecomment-1559791290
> As described [here](https://github.com/facebook/hermes/blob/e2f48291e4635b230c1267a77c05ea8efe216725/doc/Features.md#miscellaneous-incompatibilities), Hermes implements Promise by using the Promise polyfill https://github.com/then/promise, which unfortunately isn't fully compliant. This was done to ensure compatibility with React Native, which was already using it before Hermes.

...and that polyfill uses naïve functions: https://github.com/then/promise/blob/9d5851d7adefde580589a041ebfcbd030d39f6fd/src/core.js#L54
> ```js
> module.exports = Promise;
> 
> function Promise(fn) {
>   …
> }
> …
> Promise.prototype.then = function(onFulfilled, onRejected) {
> };
> …
> ```

Reply:
> set them to undefined

unfortunately we can't cauterize `.caller` or .`arguments` on Hermes (like `.prototype`) for any intrinsic

i.e. `obj[prop]`, `obj[prop] = undefined`, `obj.caller`, `Promise.arguments`, `Promise.all.caller`, etc

hit us with [`Uncaught TypeError: Restricted in strict mode`](https://github.com/search?q=repo%3Afacebook%2Fhermes%20%22Restricted%20in%20strict%20mode%22&type=code)

Reply:
how about we handle this in the permit on Hermes (similar to [earlier](https://github.com/endojs/endo/pull/2563/files#diff-f3e4581684a41bfc17458a0453b40fd9c234e02c7533a36bff380963ec7ccc40))

https://github.com/endojs/endo/pull/2655/commits/49f0f41d5d57c5080e4d78a0d1670f6667ab7a31

then remaining prototypes are successfully cauterized 🔥 too

<details>
  <summary>Hermes VM output (fixed via permits.js)</summary>

```console
➜  endo git:(ses-hermes) ✗ yarn build:hermes && yarn test:hermes
-- Building 'hermes' version of SES --
Bundle size: 448750 bytes
Copied ./types.d.ts to ./dist/types.d.cts
Concatenating: dist/ses-hermes.cjs + test/_hermes-smoke.js
Generated: test/_hermes-smoke-dist.js
Executing: test/_hermes-smoke-dist.js on Hermes compiler
test/_hermes-smoke-dist.js:13021:27: warning: the variable "Compartment" was not declared in function "?anon_0_?anon_0_testCompartmentHooks"
  const compartment = new Compartment({}, {}, { resolveHook, importHook });
                          ^~~~~~~~~~~
test/_hermes-smoke-dist.js:13029:3: warning: the variable "assert" was not declared in function "?anon_0_?anon_0_testCompartmentHooks"
  assert(module);
  ^~~~~~
test/_hermes-smoke-dist.js:3828:12: warning: the variable "AggregateError" was not declared in anonymous function " 119#"
if( typeof AggregateError!==  'undefined') {
           ^~~~~~~~~~~~~~
test/_hermes-smoke-dist.js:7065:5: warning: the variable "console" was not declared in function "getOwnPropertyDescriptor"
    console.warn(
    ^~~~~~~
test/_hermes-smoke-dist.js:8625:5: warning: the variable "harden" was not declared in arrow function "makeCausalConsoleFromLogger"
    harden(baseConsole);
    ^~~~~~
test/_hermes-smoke-dist.js:12992:3: warning: the variable "lockdown" was not declared in arrow function "testLockdown"
  lockdown();
  ^~~~~~~~
Generated: test/_hermes-smoke-dist.hbc
Hermes compiler done
Executing generated bytecode file on Hermes VM
Skipping assertDirectEvalAvailable
Removing lockdown.prototype
Tolerating undeletable lockdown.prototype === undefined
Removing harden.prototype
Tolerating undeletable harden.prototype === undefined
Removing %InitialGetStackString%.prototype
Tolerating undeletable %InitialGetStackString%.prototype === undefined
SES Removing unpermitted intrinsics
  Removing intrinsics.Promise._l
  Removing intrinsics.Promise._m
  Removing intrinsics.Promise._n
  Removing intrinsics.Promise.resolve.prototype
  Tolerating undeletable intrinsics.Promise.resolve.prototype === undefined
  Removing intrinsics.Promise.all.prototype
  Tolerating undeletable intrinsics.Promise.all.prototype === undefined
  Removing intrinsics.Promise.reject.prototype
  Tolerating undeletable intrinsics.Promise.reject.prototype === undefined
  Removing intrinsics.Promise.race.prototype
  Tolerating undeletable intrinsics.Promise.race.prototype === undefined
  Removing intrinsics.lockdown.prototype
  Tolerating undeletable intrinsics.lockdown.prototype === undefined
  Removing intrinsics.harden.prototype
  Tolerating undeletable intrinsics.harden.prototype === undefined
  Removing intrinsics.%SharedDate%.now.prototype
  Tolerating undeletable intrinsics.%SharedDate%.now.prototype === undefined
  Removing intrinsics.%InitialGetStackString%.prototype
  Tolerating undeletable intrinsics.%InitialGetStackString%.prototype === undefined
  Removing intrinsics.%InitialError%.stackTraceLimit<get>.prototype
  Tolerating undeletable intrinsics.%InitialError%.stackTraceLimit<get>.prototype === undefined
  Removing intrinsics.%InitialError%.stackTraceLimit<set>.prototype
  Tolerating undeletable intrinsics.%InitialError%.stackTraceLimit<set>.prototype === undefined
  Removing intrinsics.%InitialError%.captureStackTrace.prototype
  Tolerating undeletable intrinsics.%InitialError%.captureStackTrace.prototype === undefined
  Removing intrinsics.%InitialError%.prepareStackTrace<get>.prototype
  Tolerating undeletable intrinsics.%InitialError%.prepareStackTrace<get>.prototype === undefined
  Removing intrinsics.%InitialError%.prepareStackTrace<set>.prototype
  Tolerating undeletable intrinsics.%InitialError%.prepareStackTrace<set>.prototype === undefined
  Removing intrinsics.%SharedError%.stackTraceLimit<get>.prototype
  Tolerating undeletable intrinsics.%SharedError%.stackTraceLimit<get>.prototype === undefined
  Removing intrinsics.%SharedError%.stackTraceLimit<set>.prototype
  Tolerating undeletable intrinsics.%SharedError%.stackTraceLimit<set>.prototype === undefined
  Removing intrinsics.%SharedError%.prepareStackTrace<get>.prototype
  Tolerating undeletable intrinsics.%SharedError%.prepareStackTrace<get>.prototype === undefined
  Removing intrinsics.%SharedError%.prepareStackTrace<set>.prototype
  Tolerating undeletable intrinsics.%SharedError%.prepareStackTrace<set>.prototype === undefined
  Removing intrinsics.%SharedError%.captureStackTrace.prototype
  Tolerating undeletable intrinsics.%SharedError%.captureStackTrace.prototype === undefined
  Removing intrinsics.%SharedMath%.random.prototype
  Tolerating undeletable intrinsics.%SharedMath%.random.prototype === undefined
  Removing intrinsics.%NumberPrototype%.toLocaleString.prototype
  Tolerating undeletable intrinsics.%NumberPrototype%.toLocaleString.prototype === undefined
  Removing intrinsics.%PromisePrototype%.then.prototype
  Tolerating undeletable intrinsics.%PromisePrototype%.then.prototype === undefined
  Removing intrinsics.%PromisePrototype%.catch.prototype
  Tolerating undeletable intrinsics.%PromisePrototype%.catch.prototype === undefined
  Removing intrinsics.%PromisePrototype%.finally.prototype
  Tolerating undeletable intrinsics.%PromisePrototype%.finally.prototype === undefined
  Removing intrinsics.%StringPrototype%.localeCompare.prototype
  Tolerating undeletable intrinsics.%StringPrototype%.localeCompare.prototype === undefined
  Removing intrinsics.%FunctionPrototype%.toString.prototype
  Tolerating undeletable intrinsics.%FunctionPrototype%.toString.prototype === undefined
  Removing intrinsics.%CompartmentPrototype%.globalThis<get>.prototype
  Tolerating undeletable intrinsics.%CompartmentPrototype%.globalThis<get>.prototype === undefined
  Removing intrinsics.%CompartmentPrototype%.name<get>.prototype
  Tolerating undeletable intrinsics.%CompartmentPrototype%.name<get>.prototype === undefined
  Removing intrinsics.%CompartmentPrototype%.evaluate.prototype
  Tolerating undeletable intrinsics.%CompartmentPrototype%.evaluate.prototype === undefined
  Removing intrinsics.%CompartmentPrototype%.module.prototype
  Tolerating undeletable intrinsics.%CompartmentPrototype%.module.prototype === undefined
  Removing intrinsics.%CompartmentPrototype%.importNow.prototype
  Tolerating undeletable intrinsics.%CompartmentPrototype%.importNow.prototype === undefined
Hermes VM done
Hermes tests complete
Removing: test/_hermes-smoke-dist.js
Removing: test/_hermes-smoke-dist.hbc
```

</details>